docs/usage/locators.md

```diff
@@ -166,6 +166,30 @@ element(by.control({
   }
 }));
 ```
+### Common usecases
+
+#### Working with responsive toolbars
+A responsive toolbar may have overflowing content, depending on the screen size.
+This content will be moved to a popover which is open by pressing a toggle button in the toolbar.
+A toggle button will be shown only when there is overflowing content. This is a problem for tests because they should only try to press the button when it's visible and interactable. One way to solve this is to always start the application under test with with a fixed screen size. Another way is to first look for any toggle button - with no restriction on visibility, and then press on it only if it exists:
+```javascript
+var button = element(by.control({
+  controlType: "sap.m.ToggleButton",
+  ancestor: {
+    viewNamespace: "appUnderTest.view.",
+    viewName : "Main",
+    id: "toolbar-overflow"
+  }
+}));
+
+button.getCssValue("visibility").then(function (visibility) {
```

Comment:
checking for visibility аттрубте is very tricky, depending on a catch block is a code smell, let's find a better way. Something like finding all elements that match a selector and evaluating the condition if exactly one was found.

Reply:
we want to click on the element if it's found, and do nothing otherwise, so we don't need a catch block. an assertion here would be nice for debugging but not required

Reply:
I delved a little bit in the docs of Protractor and it seems the correct way that does not involve catch block is isPresent().